src/modules/google-organization-kp/includes/Publisher_Service.php

```diff
@@ -0,0 +1,259 @@
+		$publisher_entity = $this->wl_entity_service->get( $publisher_id );
+		$publisher_logo   = $this->wl_publisher_service->get_publisher_logo( $publisher_id );
+
+		$data['id']          = $publisher_id;
+		$data['name']        = $publisher_post->post_title;
+		$data['type']        = $publisher_entity['label'];
+		$data['description'] = $publisher_entity['description'];
+		$data['alt_name']    = $this->wl_configuration_service->get_alternate_name();
+		$data['legal_name']  = get_post_meta( $publisher_id, $schema_service::FIELD_LEGAL_NAME, true );
+
+		if ( ! empty( $publisher_logo ) ) {
+			$data['image'] = $publisher_logo['url'];
+		}
+
+		$data['same_as']       = get_post_meta( $publisher_id, $schema_service::FIELD_SAME_AS, false );
+		$data['address']       = get_post_meta( $publisher_id, $schema_service::FIELD_ADDRESS, true );
+		$data['locality']      = get_post_meta( $publisher_id, $schema_service::FIELD_ADDRESS_LOCALITY, true );
+		$data['region']        = get_post_meta( $publisher_id, $schema_service::FIELD_ADDRESS_REGION, true );
+		$data['country']       = get_post_meta( $publisher_id, $schema_service::FIELD_ADDRESS_COUNTRY, true );
+		$data['postal_code']   = get_post_meta( $publisher_id, $schema_service::FIELD_ADDRESS_POSTAL_CODE, true );
+		$data['telephone']     = get_post_meta( $publisher_id, $schema_service::FIELD_TELEPHONE, true );
+		$data['email']         = get_post_meta( $publisher_id, $schema_service::FIELD_EMAIL, true );
+		$data['no_employees']  = get_post_meta( $publisher_id, $schema_service::FIELD_NO_OF_EMPLOYEES, true );
+		$data['founding_date'] = get_post_meta( $publisher_id, $schema_service::FIELD_FOUNDING_DATE, true );
+		$data['iso_6523']      = get_post_meta( $publisher_id, $schema_service::FIELD_ISO_6523_CODE, true );
+		$data['naics']         = get_post_meta( $publisher_id, $schema_service::FIELD_NAICS, true );
+		$data['global_loc_no'] = get_post_meta( $publisher_id, $schema_service::FIELD_GLOBAL_LOCATION_NO, true );
+		$data['vat_id']        = get_post_meta( $publisher_id, $schema_service::FIELD_VAT_ID, true );
+		$data['tax_id']        = get_post_meta( $publisher_id, $schema_service::FIELD_TAX_ID, true );
+
+		return $data;
+	}
+
+	/**
+	 * Set the Organization data.
+	 *
+	 * @param array $params The parameters sent via POST.
+	 *
+	 * @return array Saved data.
+	 *
+	 * @throws Exception Throws exception if something went wrong.
+	 *
+	 * @since 3.53.0
+	 */
+	public function save( $params ) {
+		// Break out if no parameters were provided.
+		if ( empty( $params ) ) {
+			throw new Exception( 'No parameters provided.' );
+		}
+
+		// Valid Publisher types.
+		$wl_publisher_service = $this->wl_publisher_service;
+		$publisher_types = array_values( $wl_publisher_service::VALID_PUBLISHER_TYPES );
+
+		// Try to get the Publisher
+		$publisher_id = $this->wl_configuration_service->get_publisher_id();
+
+		// Publisher doesn't exist, create one.
+		if ( ! isset( $publisher_id ) || $publisher_id === '(none)' ) {
```

Comment:
Maybe we can check whether $publisher_id is number instead of `(none)`. `(none)` seems unreliable.